L1Trigger/Phase2L1Jets/plugins/L1SeedConeJetProducer.cc

```diff
@@ -0,0 +1,100 @@
+        return a + b.pt();
+    };
+
+    // Sum the pt
+    float pt = std::accumulate(parts.begin(), parts.end(), 0., sumpt);
+
+    // pt weighted d eta
+    std::vector<float> pt_deta;
+    pt_deta.resize(parts.size());
+    std::transform(parts.begin(), parts.end(), pt_deta.begin(), [&seed](const l1t::PFCandidate &part){
+            return part.pt() * (part.eta() - seed.eta());});
+    // Accumulate the pt weighted etas. Init to the seed eta, start accumulating at begin()+1 to skip seed
+    float eta = std::accumulate(pt_deta.begin()+1, pt_deta.end(), seed.eta());
+
+    // pt weighted d phi
+    std::vector<float> pt_dphi;
+    pt_dphi.resize(parts.size());
+    std::transform(parts.begin(), parts.end(), pt_dphi.begin(), [&seed](const l1t::PFCandidate &part){
+            return part.pt() * (part.phi() - seed.phi());});
+    // Accumulate the pt weighted phis. Init to the seed phi, start accumulating at begin()+1 to skip seed
+    float phi = std::accumulate(pt_dphi.begin()+1, pt_dphi.end(), seed.phi());
+
+    PFJet jet(pt, eta, phi);
+
+    return jet;
+}
+
+L1SeedConeJetProducer::L1SeedConeJetProducer(const edm::ParameterSet& cfg) : 
+    _coneSize( cfg.getParameter<double>("coneSize")),
+    _nJets( cfg.getParameter<int>("nJets")),
+    _l1PFToken( consumes<vector<l1t::PFCandidate>>(cfg.getParameter<edm::InputTag>("L1PFObjects")))
+{
+  produces< PFJetCollection >( "L1SeedConeJets" ).setBranchAlias("L1SeedConeJets");
+}
+
+void L1SeedConeJetProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+    std::unique_ptr<PFJetCollection> newPFJetCollection(new PFJetCollection);
+
+    edm::Handle<  l1t::PFCandidateCollection > l1PFCandidates;
+    iEvent.getByToken( _l1PFToken, l1PFCandidates);
+    l1t::PFCandidateCollection work;
+
+    std::copy((*l1PFCandidates).begin(), (*l1PFCandidates).end(), std::back_inserter(work));
+
+    std::sort(work.begin(), work.end(), [](l1t::PFCandidate i,l1t::PFCandidate j){return(i.pt() > j.pt());});   
+
+    std::vector<l1t::PFJet> jets;
+    jets.resize(_nJets);
+
+    bool done = false;
+    int nJet = 0;
+ 
+    //for(int nJet = 0; nJet < _nJets; nJet++){
+    while(!done){
+        // Take the first (highest pt) candidate as a seed
+        l1t::PFCandidate seed = *work.begin();
+        // Get the particles within a _coneSize of the seed
+        std::vector<l1t::PFCandidate> particlesInCone;
+        std::copy_if(work.begin(), work.end(), std::back_inserter(particlesInCone), [&](const l1t::PFCandidate &part){
+                return reco::deltaR<l1t::PFCandidate,l1t::PFCandidate>(seed, part) <= _coneSize;});
+        jets.push_back(makeJet(particlesInCone));
+        // remove the clustered particles
+        work.erase(std::remove_if(work.begin(), work.end(), [&](const l1t::PFCandidate &part){
+                return reco::deltaR<l1t::PFCandidate,l1t::PFCandidate>(seed, part) <= _coneSize;}), work.end());
+        done = (nJet >= _nJets) || work.size() == 0;
```

Comment:
`nJet` &rarr; `++nJet` ?  
otherwise, I don't see where you increment `nJets`

still, maybe it's cleaner just do change the the loop to 
````
while(!work.empty() && jets.size() < _nJets) {
   ...
} 
````